### PR TITLE
Gutenberg - Add quote block v2 flag

### PR DESCRIPTION
**Pull request by geriux**
- **Gutenberg Mobile PR**: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4744
- **Gutenberg PR**: https://github.com/WordPress/gutenberg/pull/40133
- **WordPressKit iOS PR**: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/493

Adds a Gutenberg feature flag for the Quote V2 block.

To test check the Gutenberg PR description.

## Regression Notes
1. Potential unintended areas of impact
Block editor

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
None, this is a temporary flag while we transition into the new Quote block V2. It'll be removed once both web and mobile have released it.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Models/BlockEditorSettings+GutenbergEditorSettings.swift [hunk 1]
```diff
@@ -81,7 +85,15 @@ extension BlockEditorSettings {
                                                          type: .experimentalFeatures,
                                                          order: 0,
                                                          context: context)
+        let quoteKey = BlockEditorExperimentalFeatureKeys.quoteBlockV2.rawValue
+        let quoteRefactor = BlockEditorSettingElement(name: quoteKey,
+                                                         value: "\(remoteSettings.quoteBlockV2)",
```
Comment (mchowning) [inline WordPress/Classes/Models/BlockEditorSettings+GutenbergEditorSettings.swift:90]: I'm sure this is just me not knowing Swift very well 🙂  .. why do we need to escape the opening `(`, but not the closing `)`?

**Review by mchowning — APPROVED** [review]
Code looks good and this works well. Thanks!

File: WordPress/Classes/Models/BlockEditorSettings+GutenbergEditorSettings.swift (hunk 1 above)
Comment (geriux) [reply WordPress/Classes/Models/BlockEditorSettings+GutenbergEditorSettings.swift:90]: It's called [string interpolation](https://docs.swift.org/swift-book/LanguageGuide/StringsAndCharacters.html#ID292) I had to investigate this as well while I was using another PR as a reference to create this one 😅

Comment (mchowning) [reply WordPress/Classes/Models/BlockEditorSettings+GutenbergEditorSettings.swift:90]: Ah, it's not escaping, I get it. Thanks for explaining! 🙇

**Pull request merged**